backend/service/aws/dynamodb.go

```diff
@@ -57,3 +86,60 @@ func newProtoForGlobalSecondaryIndex(index types.GlobalSecondaryIndexDescription
 		ProvisionedThroughput: currentCapacity,
 	}
 }
+
+func isValidIncrease(client *regionalClient, current *types.ProvisionedThroughputDescription, target types.ProvisionedThroughput) error {
+	// check for targets that are lower than current (can't scale down)
+	if *current.ReadCapacityUnits > *target.ReadCapacityUnits {
+		return status.Errorf(codes.FailedPrecondition, fmt.Sprintf("Target read capacity [%d] is lower than current capacity [%d]", *target.ReadCapacityUnits, *current.ReadCapacityUnits))
+	}
+	if *current.WriteCapacityUnits > *target.WriteCapacityUnits {
+		return status.Errorf(codes.FailedPrecondition, fmt.Sprintf("Target write capacity [%d] is lower than current capacity [%d]", *target.WriteCapacityUnits, *current.WriteCapacityUnits))
+	}
+
+	// check for targets that exceed max limits
+	if *target.ReadCapacityUnits > client.dynamodbCfg.ScalingLimits.MaxReadCapacityUnits {
+		return status.Errorf(codes.FailedPrecondition, fmt.Sprintf("Target read capacity exceeds maximum allowed limits [%d]", client.dynamodbCfg.ScalingLimits.MaxReadCapacityUnits))
+	}
+	if *target.WriteCapacityUnits > client.dynamodbCfg.ScalingLimits.MaxWriteCapacityUnits {
+		return status.Errorf(codes.FailedPrecondition, fmt.Sprintf("Target write capacity exceeds maximum allowed limits [%d]", client.dynamodbCfg.ScalingLimits.MaxWriteCapacityUnits))
+	}
+
+	// check for increases that exceed max increase scale
+	if (float32(*target.ReadCapacityUnits / *current.ReadCapacityUnits)) > client.dynamodbCfg.ScalingLimits.MaxScaleFactor {
+		return status.Errorf(codes.FailedPrecondition, fmt.Sprintf("Target read capacity exceeds the scale limit of [%.1f]x current capacity", client.dynamodbCfg.ScalingLimits.MaxScaleFactor))
+	}
+	if (float32(*target.WriteCapacityUnits / *current.WriteCapacityUnits)) > client.dynamodbCfg.ScalingLimits.MaxScaleFactor {
+		return status.Errorf(codes.FailedPrecondition, fmt.Sprintf("Target write capacity exceeds the scale limit of [%.1f]x current capacity", client.dynamodbCfg.ScalingLimits.MaxScaleFactor))
+	}
+	return nil
+}
+
+func (c *client) UpdateTableCapacity(ctx context.Context, region string, tableName string, targetTableRcu int64, targetTableWcu int64) error {
+	cl, err := c.getRegionalClient(region)
+	if err != nil {
+		return err
+	}
```

Comment:
```suggestion
	if err != nil {
	        c.log.Error("unable to get regional client", zap.Error(err))
		return err
	}
```

lets make sure to log all system errors like these, same applies for other error cases throughout the pr.